algorithm/sorting-algo-python/bucket_sort.py

```diff
@@ -0,0 +1,31 @@
+def bucket_sort(arr):
+    if len(arr) == 0:
+        return arr
+
+    # Determine minimum and maximum values
+    min_value = min(arr)
+    max_value = max(arr)
+
+    # Initialize buckets
+    bucket_count = len(arr)
+    buckets = [[] for _ in range(bucket_count)]
+
+    # Distribute input array values into buckets
+    for i in range(len(arr)):
+        index = int((arr[i] - min_value) / (max_value - min_value + 1) * bucket_count)
+        buckets[index].append(arr[i])
+
+    # Sort each bucket and concatenate the result
+    sorted_array = []
+    for bucket in buckets:
+        sorted_array.extend(sorted(bucket))
+
+    return sorted_array
+
+# Example usage
+    arr = [0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434]
+    print("Original array:", arr)
+    sorted_arr = bucket_sort(arr)
+    print("Sorted array:", sorted_arr)
+
+#end of the line 
```

Comment:
By EOL we mostly refer to an empty new line at the bottom of the files.
If you do so, you no longer see the red circled minus (-) icon in GitHub file changes.
![image](https://github.com/user-attachments/assets/189ff87a-6dfd-4f7f-a635-99492a9563de)

```suggestion
    sorted_arr = bucket_sort(arr)
    print("Sorted array:", sorted_arr)

```